saas/mixins.py

```diff
@@ -650,6 +650,21 @@ def period_func(self):
         period = self.request.GET.get('period')
         return self.PERIOD_FUNC_MAP.get(period, month_periods)
 
+    @property
+    def num_periods(self):
+        if not hasattr(self, '_num_periods'):
+            num_periods = self.request.GET.get('num_periods', None)
+            try:
+                # Keeping the maximum value of num_periods to 100
```

Comment:
That's interesting. At first I thought we should use a validate method on `PeriodSerializer` but then I realized we are only using the `PeriodSerializer` for documentation purposes. We have not done this so far anywhere in the code base, but there might be a way to use serializers to check query parameters (as opposed to request body). This [reddit thread](https://www.reddit.com/r/django/comments/12rspek/in_drf_do_you_validate_your_query_params_if_so_how/?rdt=53845) has some examples.